### PR TITLE
Fixed stutter on "next" button appear

### PR DESCRIPTION
**Pull request by rauchp**
feels p smooth now 💦💦💦

### CODE REVIEW

**Comment by rauchp** [comment]
cody

Cody getting ready to nit me for putting button in /common instead of ../components/

My reasoning is that we will never use this button beyond these two screens, so rather than have it clog up our components, where I want to store commonly used <Stuff />, we have common be stuff just shared by a single subfolder

File: civilapp/auth/common/NextButtonComponent.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    constructor(props) {
+        super(props)
+        
+        this.props.showButton((shouldShow) => this.setState({ shouldShow }))
+        this.props.showLoading((isLoading) => this.setState({ isLoading }))
+
+        this.state = {
+            isLoading: this.props.loadOnInit ? true : false,
+            shouldShow: this.props.showOnInit ? true : false
```
Comment (rauchp) [inline civilapp/auth/common/NextButtonComponent.js]: you hate to see the extra props, but this is necessary so on transition it doesn't revert back to "resend code"

Comment (EphemeralX) [inline civilapp/auth/common/NextButtonComponent.js]: Just for my education, we update the state here from props just to re-render only the button right?  The whole point of passing those empty functions to be defined here is to not re-render the parent components, only this?

File: civilapp/auth/ConfirmSignInScreen.js [hunk 2]
```diff
@@ -31,7 +29,6 @@ export default class ConfirmSignInScreen extends Component {
   state = {
     phoneNumber: '',
```
Comment (EphemeralX) [inline civilapp/auth/ConfirmSignInScreen.js]: If we're using this.phoneNumber, then should we remove it from state?

File: civilapp/auth/common/NextButtonComponent.js (hunk 1 above)
Comment (rauchp) [reply civilapp/auth/common/NextButtonComponent.js]: Exactly

File: civilapp/auth/ConfirmSignInScreen.js (hunk 2 above)
Comment (rauchp) [reply civilapp/auth/ConfirmSignInScreen.js]: You right, I'll remove.

**Pull request merged**